File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/adapter/ScriptGroupsAdapter.java
package com.inspiredandroid.linuxcommandbibliotheca.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

import com.inspiredandroid.linuxcommandbibliotheca.R;
import com.inspiredandroid.linuxcommandbibliotheca.interfaces.OnListClickListener;
import com.inspiredandroid.linuxcommandbibliotheca.models.BasicGroupModel;
import com.inspiredandroid.linuxcommandbibliotheca.models.CommandChildModel;
import com.inspiredandroid.linuxcommandbibliotheca.models.CommandGroupModel;
import com.inspiredandroid.linuxcommandbibliotheca.models.ScriptGroupItem;
import com.inspiredandroid.linuxcommandbibliotheca.view.TerminalTextView;

import java.util.ArrayList;
import java.util.HashMap;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.realm.OrderedRealmCollection;
import io.realm.RealmRecyclerViewAdapter;

/**
 * Created by simon on 23.11.15.
 */
public class ScriptGroupsAdapter extends RealmRecyclerViewAdapter<BasicGroupModel, ScriptGroupsAdapter.ViewHolder> {

    public ScriptGroupsAdapter(@Nullable OrderedRealmCollection<BasicGroupModel> data, boolean autoUpdate) {
        super(data, autoUpdate);
    }

    private OnListClickListener mOnListClickListener;

    public void setOnListClickListener(OnListClickListener listener) {
        mOnListClickListener = listener;
    }

    @Override
    public ScriptGroupsAdapter.ViewHolder onCreateViewHolder(ViewGroup parent,
                                                             int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.row_scriptgroup, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, int position) {
        BasicGroupModel item = getData().get(position);

        viewHolder.name.setText(item.getTitle());
        viewHolder.icon.setImageResource(item.getIconResourceId());
    }

    class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        @BindView(R.id.row_scriptgroup_tv_title)
        TextView name;
        @BindView(R.id.row_scriptgroup_iv_icon)
        ImageView icon;

        ViewHolder(View view) {
            super(view);
            ButterKnife.bind(this, view);
            view.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            if (mOnListClickListener != null) {
                mOnListClickListener.onClick(getData().get(getAdapterPosition()).getId());
            }
        }
    }
}

File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/adapter/ScriptGroupsAdapter.kt
package com.inspiredandroid.linuxcommandbibliotheca.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.inspiredandroid.linuxcommandbibliotheca.R
import com.inspiredandroid.linuxcommandbibliotheca.interfaces.OnListClickListener
import com.inspiredandroid.linuxcommandbibliotheca.models.BasicGroupModel
import io.realm.OrderedRealmCollection
import io.realm.RealmRecyclerViewAdapter
import kotlinx.android.synthetic.main.row_scriptgroup.view.*

/**
 * Created by simon on 23.11.15.
 */
class ScriptGroupsAdapter(data: OrderedRealmCollection<BasicGroupModel>?, autoUpdate: Boolean) : RealmRecyclerViewAdapter<BasicGroupModel, ScriptGroupsAdapter.ScriptViewHolder>(data, autoUpdate) {

    private var mOnListClickListener: OnListClickListener? = null

    fun setOnListClickListener(listener: OnListClickListener) {
        mOnListClickListener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup,
                                    viewType: Int): ScriptViewHolder {
        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.row_scriptgroup, parent, false)
        return ScriptViewHolder(v)
    }

    override fun onBindViewHolder(viewHolder: ScriptViewHolder, position: Int) {
        val item = data!![position]

        viewHolder.bind(item)
    }

    inner class ScriptViewHolder(view: View) : RecyclerView.ViewHolder(view), View.OnClickListener {

        init {
            view.setOnClickListener(this)
        }

        fun bind(item: BasicGroupModel) {
            itemView.row_scriptgroup_tv_title.text = item.title
            itemView.row_scriptgroup_iv_icon.setImageResource(item.iconResourceId)
        }

        override fun onClick(view: View) {
            if (mOnListClickListener != null) {
                mOnListClickListener!!.onClick(data!![adapterPosition].id)
            }
        }
    }
}
